components/cors-mgt/org.wso2.carbon.identity.cors.mgt.core/src/main/java/org/wso2/carbon/identity/cors/mgt/core/CORSManagementService.java

```diff
@@ -41,26 +42,52 @@
      * Set the CORS Origins for a tenant. This method replaces any existing Origins.
      *
      * @param tenantDomain The tenant domain.
-     * @param corsOrigins  A list of CORS Origins to be set.
+     * @param appId        The  application ID that the CORS origin(s) belongs to. Should be {@code null} if the CORS
+     *                     origin(s) belong to the tenant level.
+     * @param origins      A list of CORS origins to be set.
      * @throws CORSManagementServiceException
      */
-    void setCORSOrigins(String tenantDomain, List<CORSOrigin> corsOrigins) throws CORSManagementServiceException;
+    void setCORSOrigins(String tenantDomain, String appId, List<String> origins) throws CORSManagementServiceException;
 
     /**
      * Add the CORS Origin(s) to the existing CORS Origin list of the tenant.
      *
      * @param tenantDomain The tenant domain.
-     * @param corsOrigins  A list of CORS Origins to be added.
+     * @param appId        The  application ID that the CORS origin(s) belongs to. Should be {@code null} if the CORS
+     *                     origin(s) belong to the tenant level.
+     * @param origins      A list of CORS origins to be added.
      * @throws CORSManagementServiceException
      */
-    void addCORSOrigins(String tenantDomain, List<CORSOrigin> corsOrigins) throws CORSManagementServiceException;
+    void addCORSOrigins(String tenantDomain, String appId, List<String> origins) throws CORSManagementServiceException;
 
     /**
      * Delete the CORS Origin(s) from the existing CORS Origin list of the tenant.
      *
      * @param tenantDomain The tenant domain.
-     * @param corsOrigins  A list of CORS Origins to be deleted.
+     * @param appId        The  application ID that the CORS origin(s) belongs to. Should be {@code null} if the CORS
+     *                     origin(s) belong to the tenant level.
+     * @param originIds    A list of CORS origin IDs to be deleted.
      * @throws CORSManagementServiceException
      */
-    void deleteCORSOrigins(String tenantDomain, List<CORSOrigin> corsOrigins) throws CORSManagementServiceException;
+    void deleteCORSOrigins(String tenantDomain, String appId, List<String> originIds)
```

Comment:
Any reasons why we went with IDs instead of the origin itself here?

To me, it might be a bit confusing when we compare the addCORSOrigins() and deleteCORSOrigins() method.

Reply:
This should be fine IMO as we follow the similar approach in most of the other APIs? it gives the origin url and the id from the `getCORSOrigins()` request. So the user can use that resource id to delete the resource.

Reply:
Since the REST API is taking IDs for delete operations, I thought it might be straightforward to use them here.